### DISCUSSION TITLE
Getting random position at fixed distance

### DISCUSSION THREAD

**Discussion by arush-garg** [community/community#67699] [Programming Help] [answered]
### Body

I am trying to develop a simulation where there is the Sun in the centre, with planets orbiting around.

For initialising the planets, I need to have them at a fixed distance from the sun (orbital radius), but at random coordinates. How can I go about doing this?

I am using pygame for the graphics.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Abhiram-J** [comment]
You can use the random module in python for getting a random decimal number between 0 & 1.
```
import random
randNum = random.random()
```
If you want to get an integer between a & b, you can use the randint() function

```
import random
randNum = random.randint(a,b)
```

**Reply by arush-garg** [reply]
Thank you for your response. I know about the randint() function, but how do I set the range such that it is circular, around the point?

**Reply by Abhiram-J** [reply, marked as answer]
Inorder to draw a circle, you can use the function 
```
pygame.draw.circle(screen, color_of_circle, (a,b),r)
```
This will draw a circle with a radius of r pixels at the position (a, b) on the window.

Also now we want the planet to move around this circle you can use something similar to the below code

```
circle_center = (400, 300)
circle_radius = 100

point_radius = 5
rotation_angle = 0  # Initial rotation angle

def calculate_point_position(angle):
    x = circle_center[0] + circle_radius * math.cos(math.radians(angle))
    y = circle_center[1] + circle_radius * math.sin(math.radians(angle))
    return (int(x), int(y))

    rotation_angle += 1
    point_position = calculate_point_position(rotation_angle)

    pygame.draw.circle(screen, red, circle_center, circle_radius, 2)
    pygame.draw.circle(screen, red, point_position, point_radius)
    pygame.display.update()
```

**Reply by arush-garg** [reply]
Thanks, this is very useful!
